docs/jest-integration.md

```diff
@@ -0,0 +1,312 @@
+The Jest integration provides two modes of execution, which will be explained in
+detail further down on this page.
+
+- **Fuzzing Mode**: Fuzzing a function through a Jest test.
+- **Regression Mode**: Using initially provided seeds and inputs of found
+  problems to execute the Jest test with.
+
+## Setting up the Jazzer.js Jest integration
+
+Jest enables the execution of tests by third party tools through custom test
+runners. Jazzer.js provides such a runner in the `@jazzer.js/jest-runner`
+package.
+
+To use the integration add a `dev-dependency` to `@jazzer.js/jest-runner` to
+your project.
+
+```shell
+npm install --save-dev @jazzer.js/jest-integration
+```
+
+This will install the custom Jest runner along with all other required Jazzer.js
+dependencies.
+
+For Jest to pick up the custom fuzz test runner, it has to be added to the Jest
+configuration in the `package.json` or `jest.config.js`, as described on the
+[Configuring Jest](https://jestjs.io/docs/configuration) documentation page. The
+following snippet shows how to add the Jazzer.js fuzz test runner to the Jest
+configuration in `package.json`, so that all files with the suffix `.fuzz.js`
+will be executed by it. Other tests are still executed by the default test
+runner.
+
+```json
+{
+	"name": "jest-integration-example",
+	"scripts": {
+		"test": "jest",
+		"fuzz": "JAZZER_FUZZ=1 jest"
+	},
+	"devDependencies": {
+		"@jazzer.js/jest-runner": "1.1.0",
+		"jest": "29.3.1"
+	},
+	"jest": {
+		"projects": [
+			{
+				"displayName": "test"
+			},
+			{
+				"runner": "@jazzer.js/jest-runner",
+				"displayName": {
+					"name": "Jazzer.js",
+					"color": "cyan"
+				},
+				"testMatch": ["<rootDir>/**/*.fuzz.js"]
+			}
+		]
+	}
+}
+```
+
+Further fuzzer configuration can be specified in `.jazzerjsrc.json` in the
+project directory using the following format. These are the same properties
+which can be specified through the CLI client.
+
+```json
+{
+	"includes": ["*"],
+	"excludes": ["node_modules"],
+	"customHooks": [],
+	"fuzzerOptions": [],
+	"sync": false
+}
+```
+
+## Writing a Jest fuzz tests
+
+To create a fuzz test the `fuzz` function on Jest's `test` and `it` can be used.
+
+`fuzz` expects similar parameters as normal Jest tests but provide fuzzer
+generated input as a first parameter to the test function. As expected the input
+generation is guided by coverage and other execution feedback, added via code
+instrumentation.
+
+The passed in fuzzer input is of type `Buffer`, a subclass of `Uint8Array`, and
+can be used to create needed parameters for the actual code under test, so that
+the fuzzer can detect the usage of parts of the input and mutate them in the
+next iterations, to reach new code paths.
+
+The resulting code looks very similar to normal Jest tests.
+
+```javascript
+describe("My describe", () => {
+	it.fuzz("My fuzz test", (data) => {
+		target.fuzzMe(data);
+	});
+});
+```
+
+However, the provided `Buffer` is not the nicest abstraction to work with. For
```

Comment:
IMO: This again, reads like a duplication of the other `README.md`. We should either re-write parts or make sure we add more value to the parts we duplicate here

Reply:
True, but let's extract out a dedicated `FuzzedDataProvider` section in another PR.